framework/src/multiapps/MultiApp.C

```diff
@@ -482,6 +482,20 @@ MultiApp::restore()
   }
 }
 
+void
+MultiApp::restoreSubApps()
+{
+  for (unsigned int i = 0; i < _apps.size(); i++)
+  {
+    for (auto & sub_app :
+         _apps[i]->getExecutioner()->feProblem().getMultiAppWarehouse().getObjects())
+      if (sub_app->needsRestoration())
+        sub_app->restore();
```

Comment:
So if you call `restore` on a sub-app, then that restores all sub-sub-apps as well?

Reply:
That's my understanding at least ;) 
It seems to me that any app restores everything under it if it is restored.